### PR TITLE
Fixes #34878 - enable Red Hat repositories to resolve dependencies if…

### PR DESCRIPTION
**Pull request by upadhyeammit**
… any

### CODE REVIEW

**Comment by upadhyeammit** [comment]
There could be systems not having psql/nftables packages installed and newer version of foreman-maintain requires these as dependencies.
I am yet to cherry pick https://projects.theforeman.org/issues/34855 for 6.11 and I think we need this change as well.

File: definitions/scenarios/self_upgrade.rb [hunk 1]
```diff
@@ -79,6 +79,20 @@ def use_rhsm?
 
       true
     end
+
+    def main_rh_repos
```
Comment (evgeni) [inline definitions/scenarios/self_upgrade.rb:83]: can't we use the same method from the `downstream` concern?

Comment (upadhyeammit) [reply definitions/scenarios/self_upgrade.rb:83]: > can't we use the same method from the `downstream` concern?

I felt including whole concern for a method which does not depend anything from downstream concern is too much to include, but yes we can

Comment (evgeni) [reply definitions/scenarios/self_upgrade.rb]: I can see that, but it also means that we have to places to maintain those repo lists, and you know how that will end ;-)

**Review by evgeni — APPROVED** [review]

**Pull request merged**